tests/lib/rules/no-unused-vars.js

```diff
@@ -333,6 +345,21 @@ ruleTester.run("no-unused-vars", rule, {
             ]
         },
 
+        // Using object rest for variable omission
+        {
+            code: "const data = { type: 'coords', x: 1, y: 2 };\nconst { type } = data;",
```

Comment:
Yeah I wasn't sure how necessary this was, but thought it would be nice to have a destructuring case without the rest sibling just for sanity. Need to change the comment for sure tho. Copy pasta.